src/Text/Pandoc/Readers/JATS.hs

```diff
@@ -616,24 +623,44 @@ parseInline (Elem e) =
         _          -> innerInlines id
    where innerInlines f = extractSpaces f . mconcat <$>
                           mapM parseInline (elContent e)
-         mathML x =
-            case readMathML . showElement $ everywhere (mkT removePrefix) x of
-                Left _ -> mempty
-                Right m -> writeTeX m
-         formula constructor = do
```

Comment:
Split `formula` function into `inlineFormula` and `blockFormula` to deal with each representation in the native document. (Could further recycle common code, but that is scope for a further PR, for clarity leaving as is for now).